backfiller/backfiller/main.py

```diff
@@ -315,18 +325,22 @@ def backfill(self, hours, segment_order='random', recent_cutoff=60):
 
 	def run(self):
 		self.logger.info('Starting')
+		failures = 0
 
 		while not self.stopping.is_set():
 
 			try:
 				self.backfill(list_hours(self.node, self.stream, self.variants, self.start))
-				self.failures = 0 #reset failure count on a successful backfill
+				failures = 0 #reset failure count on a successful backfill
 				if not self.run_once:
 					self.stopping.wait(common.jitter(self.WAIT_INTERVAL))
 
 			except Exception:
-				self.failures += 1
-				delay = common.jitter(self.RETRY_INTERVAL * 2**self.failures)
+				failures += 1
+				if failures > MAX_RETRIES:
```

Comment:
I don't think giving up after too many retries is a good idea, here or in get nodes. we want the service to be able to recover itself and it can't do that if it stops trying.
when i said we should have a limit on the backoff i meant a cap on the retry interval after which we continue to retry every minute or whatever, not give up.